### ISSUE TITLE
Use OpenDocument instead of Word format for Modelica specification

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/ModelicaSpecification#67]
**Reported by otter on 10 Apr 2008 15:33 UTC**
Email discussion reveals that people would like to change from Word to OpenDocument format for the Modelica 3.1 specification. Reasons:

 * Word is not reliable for multiple authors using country specific Word versions
 * OpenOffice seems to be much more reliable (experience from Francesco and Martin)
 * Plug-ins/programs in both directions are available Word <> OpenDocument (for reading, not suitable for editing and exchanging).
 * An organization developing free documents and free libraries should use an open/free format for their documents

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/67

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 11 Apr 2008 09:53 UTC**

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 6 May 2008 07:04 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by christiankral on 26 May 2008 12:02 UTC**
On May 28 it was announced by Microsoft that with Service Pack 2 of MS Office 2007 ODF will be supported. It will then be possible to open, edit and save documents using ODF.

[Announcement made by Microsoft](http://www.microsoft.com/Presspass/press/2008/may08/05-21ExpandedFormatsPR.mspx)

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 28 May 2008 11:49 UTC**
This was missed at the 57th Design Meeting and should definitely decided on at the next meeting.

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 28 May 2008 11:50 UTC**

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 28 May 2008 11:51 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 10 Sep 2008 10:30 UTC**
As there was already an ongoing discussion at the meeting in Waterloo, it would be nice to get a list of the **arguments** against and for the ODF approach. 

Just as an after thought, I for example, would never dare to make a change to the specification document because I simply don't own a copy of MS Word and am to cautious of using OOo to save as doc as some properties might be lost. So do I have to buy MS Office just so I can make a contribution to an open standard. This sounds **so** wrong.

**Comment by modelica-trac-importer** [comment]
**Comment by mtiller on 10 Sep 2008 13:43 UTC**
My recollection is that it came down to this...There are several people who edit the specification heavily.  It didn't make sense for the group as a whole (many of whom don't do much editing) to make a decision about how the heaviest editors should work unless they completely back the proposal.

That being said, I'm not sure where we stand with this (i.e. is the issue closed or ongoing?!?).

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 11 Sep 2008 16:28 UTC**
One issue that Francesco recalls from the discussion is that we have to make sure that you can still use Word with .odt and with no inconvenience whatsoever. 
So we should proceed with this checking out that the **conversion tools** and the **plug-ins** for MS Word 2007 really work. For this we would need some testers who have enough experiences (and access) to both MS Word 2007 and OOo to test the interoperability.

 1. According to Martin, converting the language specification document from .doc to .odt using OOo loses some formatting info, which should be then edited manually. Maybe if we first go to docx, and then open the docx in OOo this doesn't happen. Can somebody check this with Word 2007?
 2. Office 2007 eventually reads .odt (and doesn't mess it up)
 3. The Sun plugin for Office 2007 really works (and doesn't mess it up)

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 14 Nov 2008 17:36 UTC**
Some more points on this issue:

Microsoft has announced plans to [add support for a several new document formats to Office 2007](http://www.microsoft.com/Presspass/press/2008/may08/05-21ExpandedFormatsPR.mspx). The company plans to release Office 2007 SP2 [in spring of 2009](http://www.vnunet.com/vnunet/news/2228866/office-2007-sp2-due-spring), and it will add support for XPS, PDF 1.5, and ODF 1.1 files, among others.

This means that Modelica Association can finally switch to OD-Formats because:
 * it is an open, documented format
 * it is an ISO-OSI standard
 * it is supported by all major office suites, including Microsoft Word (as of spring 2009) and OpenOffice (already)
 * it is XML based, so it cannot cause crashes if corrupted, contrary to the binary .doc format

**Comment by modelica-trac-importer** [comment]
**Comment by HansOlsson on 17 Nov 2008 13:19 UTC**
Postponed until actual releases - hopefully done before Modelica spec 3.1,
but not at Design 60 meeting.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 6 Jun 2009 11:56 UTC**
Finally [Service Pack 2 for MS Office](http://support.microsoft.com/kb/953195) is available. This now supports natively.

Quote: 

----

## Overview of Service Pack 2 (SP2) improvements
## File formats
 * OpenDocument Format (ODF) support
      SP2 lets you open, edit, and save documents in version 1.1 of the ODF for  [Word](http://office.microsoft.com/en-us/word/FX100649251033.aspx), for [Excel](http://office.microsoft.com/en-us/excel/FX100646951033.aspx), and for [ PowerPoint](http://office.microsoft.com/en-us/powerpoint/FX100648951033.aspx). Users of these Office programs can now open, edit, and save files in the OpenDocument Text (\*.odt), OpenDocument Spreadsheet (\*.ods), and OpenDocument Presentations (\*.odp) formats.

----


This means we can finally use an open document for the specification without putting any burden on people who prefer/have to use MS Office. We would need a volunteer who has access to MS Office 2007 and can open the specification document and save it as odt. Ideally this should be tested **before** the [milestone:Design63 63rd Modelica Design Meeting] so we can finally decide to switch.

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 19 Nov 2009 16:03 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 19 Nov 2009 16:25 UTC**
Now that I have access to MSOffice 2007 SP2 I did a quick test with the _ModelicaSpecification31.doc file. The result: Yes MSOfice can save as odt and also open it (and in MSO it looks the same. The problem is hat OpenOffice will exit with an error when it tries to open that odt file. **Well done Microsoft!!!** :(

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 24 Nov 2009 23:09 UTC**
This was discussed shortly at [milestone:Design64] and the general conclusion was:
 1. Even if MSOffice is able to save odt (and taken that can be opened by OOo) it won't be a good idea to work on the Specification with MSO and OOo in parallel. Experiences with MSOffice (2003 & 2007) have shown incompatibility problems even within MSO's own format. 
 2. Some companies simply wouldn't have/allow MSO2007.

So the best option will be to use OOo and that people who actively work with the Specification will use OOo (which is available to everyone). But we still need feedback if this is fine for everyone (who will be working with the document).

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 24 Nov 2009 23:09 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 26 Nov 2009 17:48 UTC**
Update: Using Sun's [Sun ODF Plugin for Microsoft Office](http://www.sun.com/software/star/odf_plugin/) (compatible also with older MSOffice suites) the export to `odt` of the Specification was actually successful. As far as I could tell only minor corrections would be necessary. But those would be done anyway when working on the new document. So it's only a matter of people willing to use OpenOffice.org for the work on the Spec.

Interesting side effect: The produced PDF of Openoffice.org is only half in size and the odt is only a third in size.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 26 Nov 2009 18:06 UTC**
Added OpenDocumentFormat version of Specification in r3198. (Automatically created by the Sun ODF Plugin, no manual fine-tuning).

**Comment by modelica-trac-importer** [comment]
**Comment by HansOlsson on 15 Sep 2010 14:23 UTC**
Tried to experiment with it. One issue is that German issue of OpenOffice is not compatible with US-version; the text saved by German version is shown in an odd font in the US-version. Using the same OpenOffice version (including language) seems to resolve that issue, i.e. we will have to standarize on language-version.

The plan is to use US OpenOffice version for version 4.

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#x2e;se on 4 Dec 2014 13:31 UTC**
An alternative would be LaTeX. There exists some pretty good word to latex converters.

**Comment by modelica-trac-importer** [comment]
**Changelog removed by sjoelund&#46;se on 4 Dec 2014 13:31 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 11 Jun 2015 14:48 UTC**
This topic has been discussed extensively at milestone:Design86 and resulted in a specific rST based proposal which is now discussed in #1730. 

Closing this one as duplicate now.
